src/Middleware/HttpOverrides/src/ForwardedHeadersMiddleware.cs

```diff
@@ -165,6 +167,21 @@ public void ApplyForwarders(HttpContext context)
             entryCount = Math.Max(forwardedHost.Length, entryCount);
         }
 
+        if (_options.ForwardedHeaders.HasFlag(ForwardedHeaders.XForwardedPrefix))
+        {
+            checkPrefix = true;
+            forwardedPrefix = requestHeaders.GetCommaSeparatedValues(_options.ForwardedPrefixHeaderName);
+            if (_options.RequireHeaderSymmetry
+                && ((checkFor && forwardedFor!.Length != forwardedPrefix.Length)
+                    || (checkProto && forwardedProto!.Length != forwardedPrefix.Length)
+                    || (checkHost && forwardedHost!.Length != forwardedPrefix.Length)))
+            {
+                _logger.LogWarning(1, "Parameter count mismatch between X-Forwarded-Prefix and X-Forwarded-Host and X-Forwarded-For or X-Forwarded-Proto.");
```

Comment:
We'll need a test case that checks for this I think.

Reply:
Added test